### PR TITLE
feat: add view for allowance creation

### PR DESCRIPTION
**Pull request by alangsto**
**JIRA:** [COSMO-369](https://2u-internal.atlassian.net/browse/COSMO-369)

**Description:** This PR adds the ability to create one or more allowances via a POST request.

**Merge checklist:**
- [x] All reviewers approved
- [x] CI build is green
- [ ] Changelog record added
- [x] Documentation updated (not only docstrings)
- [x] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: edx_exams/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -1851,3 +1861,45 @@ def test_get_empty_response(self):
...
+
+        course_allowances = StudentAllowance.objects.all()
+        self.assertEqual(len(course_allowances), 3)
+
+        self.assertEqual(len(StudentAllowance.objects.filter(user_id=self.user.id)), 2)
+        self.assertEqual(len(StudentAllowance.objects.filter(extra_time_mins=45)), 3)
+
+    def test_invalid_post(self):
+        """
+        Test that 400 response is returned if serializer is invalid
```
Comment (schenedx) [inline edx_exams/apps/api/v1/tests/test_views.py:1899]: Can you add a test where none of the expected query strings are provided? No `exam_id`, no `user_id`, and no `extra_time_mins`? What about test that some querystrings are missing?

**Review by schenedx — APPROVED** [review]
👍

File: edx_exams/apps/api/v1/views.py [hunk 2]
```diff
@@ -808,3 +827,37 @@ def get(self, request, course_id):
...
+
+        if serializer.is_valid():
+            allowance_objects = [
+                StudentAllowance(
+                    user=(
+                        User.objects.get(username=allowance['username'])
+                        if allowance.get('username')
+                        else User.objects.get(email=allowance['email'])
+                    ),
+                    exam=Exam.objects.get(id=allowance['exam_id']),
```
Comment (Zacharis278) [inline edx_exams/apps/api/v1/views.py:847]: Normally I'd say don't query inside a loop but it's probably okay here. Probably a waste of complexity optimizing it. It could be worth at least commenting that we expect the number of allowances to be reasonably small.

**Review by Zacharis278 — APPROVED** [review]
👍
